src/utils/deploy.ts

```diff
@@ -0,0 +1,25 @@
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+
+/**
+ * Retrieves the deployer account based on the current network environment.
+ *
+ * @param hre - The Hardhat Runtime Environment.
+ * @returns The deployer account address or private key.
+ */
+export const getDeployerAccount = async (hre: HardhatRuntimeEnvironment) => {
+    const { deployer } = await hre.getNamedAccounts();
```

Comment:
I pushed a small change to the code. I moved the happy path outside of the branching code because the branching will only execute rarely (there are only few networks with the zksync environment)